examples/blend/main.go

```diff
@@ -0,0 +1,193 @@
+	textSpacing  float64 // Additional vertical margin for text.
+	modes        []mode
+}
+
+func NewGame() *Game {
+	g := &Game{
+		source: loadImage(blend.Source_png),
+		dest:   loadImage(blend.Dest_png),
+		gridW:  4, gridH: 3,
+		tileSpacing: 64, textSpacing: 16,
+	}
+
+	// Adding all known blend modes and their names.
+	g.modes = []mode{
+		{blend: ebiten.BlendCopy, name: "BlendCopy"},
+		{blend: ebiten.BlendSourceAtop, name: "BlendSourceAtop"},
+		{blend: ebiten.BlendSourceOver, name: "BlendSourceOver"},
+		{blend: ebiten.BlendSourceIn, name: "BlendSourceIn"},
+		{blend: ebiten.BlendSourceOut, name: "BlendSourceOut"},
+		{blend: ebiten.BlendDestination, name: "BlendDestination"},
+		{blend: ebiten.BlendDestinationAtop, name: "BlendDestinationAtop"},
+		{blend: ebiten.BlendDestinationOver, name: "BlendDestinationOver"},
+		{blend: ebiten.BlendDestinationIn, name: "BlendDestinationIn"},
+		{blend: ebiten.BlendDestinationOut, name: "BlendDestinationOut"},
+		{blend: ebiten.BlendClear, name: "BlendClear"},
+		{blend: ebiten.BlendXor, name: "BlendXor"},
+		{blend: ebiten.BlendLighter, name: "BlendLighter"},
+	}
+
+	// Adjusting the tile size for the available images.
+	g.tileSize = maxSide(g.source, g.dest)
+	g.offscreen = ebiten.NewImage(int(g.tileSize), int(g.tileSize))
+
+	return g
+}
+
+func (g *Game) Update() error {
+	return nil
+}
+
+func (g *Game) Draw(screen *ebiten.Image) {
+	screen.Fill(colornames.White)
+
+	// Get an offset for center alignment.
+	// Where sw, sh is the screen size in pixels.
+	// And gw, gh is the grid size in pixels.
+	sw, sh := float64(screen.Bounds().Dx()), float64(screen.Bounds().Dy())
+	gw, gh := g.gridW*(g.tileSize+g.tileSpacing)-g.tileSpacing, g.gridH*(g.tileSize+g.tileSpacing)
+	cx, cy := (sw-gw)/2, (sh-gh)/2
+
+	// Drawing a tilemap
+	for y := 0.; y < g.gridH; y++ {
+		for x := 0.; x < g.gridW; x++ {
+			px, py := x*(g.tileSize+g.tileSpacing), y*(g.tileSize+g.tileSpacing)
+			if y > 0 {
+				// Making a place for the text.
+				py += g.textSpacing * y
+			}
+
+			// Drawing the blend mode and it's name.
+			mode := getMode(g.modes, int(x), int(y), int(g.gridW), int(g.gridH))
+			g.drawBlendMode(screen, px+cx, py+cy, mode.blend)
+			drawCenteredText(screen, px+cx+g.tileSize/2, py+cy+g.tileSize+g.textSpacing, mode.name)
+		}
+	}
+}
+
+func (g *Game) Layout(w, h int) (int, int) {
+	return w, h
+}
+
+func (g *Game) drawBlendMode(screen *ebiten.Image, x, y float64, mode ebiten.Blend) {
+	// Copy the destination image to offscreen so as not to modify it.
+	g.offscreen.Clear()
+	g.offscreen.DrawImage(g.dest, nil)
+
+	// Select and apply blending mode.
+	op := &ebiten.DrawImageOptions{}
+	op.Blend = mode
+	g.offscreen.DrawImage(g.source, op)
+
+	// Draw the result on the passed coordinates.
+	op = &ebiten.DrawImageOptions{}
+	op.GeoM.Translate(x, y)
+	screen.DrawImage(g.offscreen, op)
+}
+
+// loadImage is a util function for loading embedded images.
+func loadImage(data []byte) *ebiten.Image {
```

Comment:
Let's return an error instead of logging in this function

Reply:
Is there a sense to return an error here?
Where then to handle it? In main and then pass into the constructor?

Reply:
Yeah. I think this makes sense. Handling error is not loadImage's responsibility.

Reply:
So yes, NewGame will return an error.

Reply:
Ok

Reply:
Fixed